gatsby-node.js

```
@@ -42,6 +42,107 @@ const getMessages = (path, language) => {
+ * If translation file exists, isContentEnglish will be false
+ * @param {*} path url path used to derive file path from
+ * @param {*} lang language abbreviation for language path
+ * @returns {{isOutdated: boolean, isContentEnglish: boolean}}
+ */
+const checkIsPageOutdated = async (path, lang) => {
+  try {
+    // Files that need index appended on the end. Ex page-index.json, page-developers-index.json, page-eth2-index.json
+    const indexFilePaths = ["", "developers", "eth2"]
+    const filePath = path.split("/").filter((text) => text !== "")
+
+    if (
+      indexFilePaths.includes(filePath[filePath.length - 1]) ||
+      filePath.length === 0
+    ) {
+      filePath.push("index")
+    }
+
+    const joinedFilepath = filePath.join("-")
+    const srcPath = `${__dirname}/src/intl/${lang}/page-${joinedFilepath}.json`
+    const englishPath = `${__dirname}/src/intl/en/page-${joinedFilepath}.json`
+
+    // If no file exists, default to english
+    if (!fs.existsSync(srcPath)) {
+      return {
+        isOutdated: true,
+        isContentEnglish: true,
+      }
+    } else {
+      const translatedData = JSON.parse(fs.readFileSync(srcPath))
+      const englishData = JSON.parse(fs.readFileSync(englishPath))
+      const translatedKeys = Object.keys(translatedData)
+      const englishKeys = Object.keys(englishData)
+
+      // Check if same amount of keys
+      if (translatedKeys.length !== englishKeys.length) {
+        return {
+          isOutdated: true,
+          isContentEnglish: false,
+        }
+      }
+
+      // Check if all the keys are the same
+      if (
+        JSON.stringify(translatedKeys.sort()) !==
+        JSON.stringify(englishKeys.sort())
+      ) {
+        return {
+          isOutdated: true,
+          isContentEnglish: false,
+        }
+      }
+
+      return {
+        isOutdated: false,
+        isContentEnglish: false,
+      }
+    }
+  } catch {
+    return true
```

I don't like this return that is absorbing all the errors and is different from the return type from the function.

Let the function throw its errors and that the consumer handles what to do with it.